implementation/src/main/java/io/smallrye/jwt/auth/principal/KeyLocationResolver.java

```diff
@@ -178,16 +180,28 @@ protected void initializeKeyContent() throws Exception {
                 ? authContextInfo.getPublicKeyContent()
                 : readKeyContent(authContextInfo.getPublicKeyLocation());
         // Try to init the verification key from the local PEM or JWK(S) content
-        verificationKey = tryAsPEMPublicKey(content, authContextInfo.getSignatureAlgorithm());
-        if (verificationKey == null) {
+        verificationKey = null;
+        if (mayBePemKey()) {
+            verificationKey = tryAsPEMPublicKey(content, authContextInfo.getSignatureAlgorithm());
+            if (isPemKey()) {
+                return;
+            }
+        }
```

Comment:
I'm wondering if the check for `isPemKey` should instead check for `verificationKey !=null`. Suppose `KeyFormat.ANY` and a PEM key is successfully parsed. Same goes for the other types below.

Reply:
@MikeEdgar In that case the existing `if (verificationKey == null ...)` guards will prevent any other attempts to load the key. And we also don't want the check to continue if `verificationKey` is null but the user asked for `PEM_KEY`. 
I can remove the existing `verificationKey == null ` guards and instead have
```
if (verificationKey != null || isPemKey()) {
return;
}
```
and do it within every other if block but I'm not sure it will make things clearer. What is your preference ?

Reply:
@sberyozkin - I got it now. I say keep it as-is.